src/services/service.ts

```diff
@@ -294,6 +294,25 @@ export function getAppChartRef(appId: number): Promise<ResponseType> {
         }
     })
 }
+
+export function getChartReferencesForAppAndEnv(appId: number, envId: number): Promise<ResponseType> {
+    const URL = `${Routes.CHART_REFERENCES_MIN}/${appId}/${envId}`;
+    return get(URL);
+}
+
+export function getAppChartRefForAppAndEnv(appId: number, envId: number): Promise<ResponseType> {
+    return getChartReferencesForAppAndEnv(appId, envId).then((response) => {
+        const { result: { chartRefs, latestEnvChartRef } } = response;
+        let selectedChartId = latestEnvChartRef;
+        let chart = chartRefs.find(chart => selectedChartId === chart.id);
```

Comment:
what if chartRefs are undefined or null as backend can throw an error.

Reply:
Please make sure that latestEnvChartRef should not be 0 or empty (in this case - chart will be undefined, which was not the case earlier) (could be problem for some corner case)

Reply:
using latestAppChartRef as backup as it will always be available at his stage

Reply:
showing a toast now in case of error but still it will break the UI